cli/azd/cmd/pipeline.go

```diff
@@ -146,14 +148,15 @@ func newPipelineConfigAction(
 		env:            env,
 		console:        console,
 		commandRunner:  commandRunner,
+		prompters:      prompters,
 	}
 
 	return pca
 }
 
 // Run implements action interface
 func (p *pipelineConfigAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	err := provisioning.EnsureEnv(ctx, p.console, p.env, p.accountManager)
+	err := p.prompters.EnsureEnv(ctx)
```

Comment:
NIT: I think long term we don't want EnsureEnv() to be on the prompters type.  The policy of values should be defined in the environment for provisioning to work feels better suited for the provisioning package.  I feel like we should still have a a provisioning.EnsureEnv and it should just take p.env and p.prompters and then only call stuff on p.prompters if we need it.

But happy with this for now, since that's yet another refactoring, and to handle that in a follow up PR.

Reply:
I'm a little worried about the directional shift here. Can we handle this separately, as it doesn't seem to directly correspond to "async deprecation"? The prompt isn't async, unless I'm misunderstanding here.

We've now introduced a new type of layer. A "prompter" layer that is composed of business logic + UI. How do you separate "prompterForProvisioning" vs "prompterForAccount"? What tradeoffs did we make here?

Reply:
The prompts here only represent the prompts for subscription, location & resource group.  These prompts are not provisioning specific and not sure why it originally ended up here.  I moved them out into a new `prompt` package to more easily support reusing this in the future as needed.

This PR was also intended to clean up the provider registrations and make all the dependencies injectable.

Reply:
In my opinion they are more closely related to environment configuration than provisioning.  These same values are required for other commands such as `azd pipeline config`